scipy/stats/_survival.py

```diff
@@ -1,38 +1,213 @@
-from dataclasses import make_dataclass
+import warnings
+from dataclasses import dataclass, field
 import numpy as np
-from ._censored_data import CensoredData
+from scipy import special, interpolate
+from scipy.stats._censored_data import CensoredData
+from scipy.stats._common import ConfidenceInterval
 
 
 __all__ = ['ecdf']
 
 
-ECDFResult = make_dataclass('ECDFResult', ['x', 'cdf', 'sf'])
+@dataclass
+class EmpiricalDistributionFunction:
+    """An empirical distribution function produced by `scipy.stats.ecdf`
+
+    Attributes
+    ----------
+    points : ndarray
+        The point estimate of the cumulative distribution function (CDF) or its
+        complement, the survival function (SF), at unique values of the sample.
+    """
+    points: np.ndarray
+    # Exclude these from __str__
+    _x: np.ndarray = field(repr=False)  # points at which function is estimated
+    _n: np.ndarray = field(repr=False)  # number "at risk"
+    _d: np.ndarray = field(repr=False)  # number of "deaths"
+    _sf: np.ndarray = field(repr=False)  # survival function for var estimate
+    _kind: str = field(repr=False)  # type of function: "cdf" or "sf"
+
+    def __init__(self, x, points, n, d, kind):
+        self.points = points
+        self._x = x
+        self._n = n
+        self._d = d
+        self._sf = points if kind == 'sf' else 1 - points
+        self._kind = kind
+        f0 = 1 if kind == 'sf' else 0  # leftmost function value
+        f1 = 1 - f0
+        # fill_value can't handle edge cases at infinity
+        x = np.insert(x, [0, len(x)], [-np.inf, np.inf])
+        y = np.insert(points, [0, len(points)], [f0, f1])
+        # `or` conditions handle the case of empty x, points
+        self._f = interpolate.interp1d(x, y, kind='previous',
+                                       assume_sorted=True)
+
+    def confidence_interval(self, confidence_level=0.95, method='Greenwood'):
+        """Compute a confidence interval around the CDF/SF point estimate
+
+        Parameters
+        ----------
+        confidence_level : float, default: 0.95
+            Confidence level for the computed confidence interval
+
+        method : str, optional
```

Comment:
Above you use `default`. We should do that here as well or change above to make it consistent.

Reply:
Yup.
```suggestion
        method : str, default: "Greenwood"
```